test-support/reference-trie/src/lib.rs

```diff
@@ -51,6 +207,32 @@ const EXTENSION_NODE_OVER: u8 = BRANCH_NODE_NO_VALUE - EXTENSION_NODE_OFFSET;
 const LEAF_NODE_LAST: u8 = EXTENSION_NODE_OFFSET - 1;
 const EXTENSION_NODE_LAST: u8 = BRANCH_NODE_NO_VALUE - 1;
 
+/*mod noext_cst {
+	pub const EMPTY_TRIE: u8 = 0;
+	pub const LEAF_NODE_OFFSET: u8 = 1;
+	pub const LEAF_NODE_BIG: u8 = 85;
+	pub const BRANCH_NODE_NO_VALUE: u8 = 86;
+	pub const BRANCH_NODE_NO_VALUE_BIG: u8 = 170;
+	pub const BRANCH_NODE_WITH_VALUE: u8 = 171;
+	pub const BRANCH_NODE_WITH_VALUE_BIG: u8 = 255;
+	pub const LEAF_NODE_OVER: u8 = LEAF_NODE_BIG - LEAF_NODE_OFFSET;
+	pub const BRANCH_NODE_WITH_VALUE_OVER: u8 = BRANCH_NODE_WITH_VALUE_BIG - BRANCH_NODE_WITH_VALUE;
+	pub const BRANCH_NODE_NO_VALUE_OVER: u8 = BRANCH_NODE_NO_VALUE_BIG - BRANCH_NODE_NO_VALUE;
+	pub const LEAF_NODE_LAST: u8 = LEAF_NODE_BIG - 1;
+	pub const BRANCH_NODE_WITH_VALUE_LAST: u8 = BRANCH_NODE_WITH_VALUE_BIG - 1;
+	pub const BRANCH_NODE_NO_VALUE_LAST: u8 = BRANCH_NODE_NO_VALUE_BIG - 1;
+}*/
+
+/// constant use with trie simplification codec
+mod s_cst {
+	pub const EMPTY_TRIE: u8 = 0;
+	pub const NIBBLE_SIZE_BOUND: usize = u16::max_value() as usize;
```

Comment:
`MAX_KEY_NIBBLES` is more appropriate. NIbble size is always 4 bits.

Reply:
codec can be use with other length of nibble size (TtrieStream cannot), I am starting to consider removing this possibility through (I did not run enough test on other trie radix).